src/nl.rs

```diff
@@ -148,50 +45,84 @@ impl CanInterface {
     /// Open CAN interface by name
     ///
     /// Similar to `open_if`, but looks up the device by name instead
-    pub fn open(ifname: &str) -> Result<CanInterface, nix::Error> {
+    pub fn open(ifname: &str) -> result::Result<Self, nix::Error> {
```

Comment:
Why the `result::Result`? It's not competing with any other type being brought into scope.

Reply:
That's what I get for commenting on a WIP PR. :P

Reply:
No, @tobz, you're right. I was going down that path, but received some feedback saying to keep it the way it is for now, and didn't clean up properly. I'm trying to find the balance of what people want to do in the next release or two.